[stable/prometheus-operator] Add release namespace to header in sync scripts

mercul3s opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:
This PR adds the release namespace to grafana dashboards and alerting rules generated from `sync_grafana_dashboards.py` and `sync_prometheus_rules.py`. This ensures that the rules and dashboards configmaps are deployed to the same namespace as prometheus and grafana.

Note: This change does not contain updates to the templates, as I was unsure if that was necessary at this point and ran into some issues with updated alertmanager rules which I don't have bandwidth to debug at this time (I've disabled them in our deployment for now). I can resubmit with updated templates if desired.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

mercul3s committed:
Add release namespace to header

This will ensure configmaps are in the correct namespace.

Signed-off-by: Mercedes Coyle <me@mercedescoyle.com>

bismarck commented on stable/prometheus-operator/hack/sync_grafana_dashboards.py:
```diff
@@ -77,6 +77,7 @@ def new_representer(dumper, data):
 kind: ConfigMap
 metadata:
   name: {{ printf "%%s-%%s" (include "prometheus-operator.fullname" $) "%(name)s" | trunc 63 | trimSuffix "-" }}
+  namespace: {{ .Release.Namespace }}
```
Could you make this `$.Release.Namespace`?

bismarck commented on stable/prometheus-operator/hack/sync_prometheus_rules.py:
```diff
@@ -117,6 +117,7 @@ def new_representer(dumper, data):
 kind: PrometheusRule
 metadata:
   name: {{ printf "%%s-%%s" (include "prometheus-operator.fullname" .) "%(name)s" | trunc 63 | trimSuffix "-" }}
+  namespace: {{ .Release.Namespace }}
```
Could you make this `$.Release.Namespace`?

bismarck reviewed (changes requested):
@mercul3s  Thanks for the PR!  Could you try regenerating the rules and dashboards with the updated scripts and include those in your PR?

mercul3s committed:
Update dashboard and rules templates

This change regenerates grafana dashboards and prometheus rules that
include the release namespace.

Signed-off-by: Mercedes Coyle <me@mercedescoyle.com>

mercul3s committed:
Version bump

Signed-off-by: Mercedes Coyle <me@mercedescoyle.com>
